### PR TITLE
Implement Aggregation.

### PR DESCRIPTION
**Pull request by dairyo**
Implement `Aggregation` class and its tests.

Related issue #296

### CODE REVIEW

**Comment by yongpingchen** [comment]
Can you fix the circleCI script to be able to check the test result.

File: ThingIFSDK/ThingIFSDK/Aggregation.swift [hunk 1]
```diff
@@ -58,12 +58,20 @@ open class Aggregation: NSCoding {
...
+    private init?(
       _ function: FunctionType,
       field: String,
       fieldType: FieldType)
     {
-        fatalError("TODO: implement me.")
+        if function == .max || function == .sum || function == .min ||
+             function == .mean {
+            if fieldType != .integer && fieldType != .decimal {
+                return nil
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/Aggregation.swift]: I don't think it is good only return nil when fieldType is wrong. It is better to check the fieldType in `ThingIFAPI`, where can return error.

Comment (dairyo) [reply ThingIFSDK/ThingIFSDK/Aggregation.swift]: @yongpingchen 
That is good but that changes behavior between Android and iOS.

What do you think about this?

**Comment by dairyo** [comment]
@yongpingchen 

> Can you fix the circleCI script to be able to check the test result.

I would like to change script in other PR. Could you allow this?

**Comment by yongpingchen** [comment]
> I would like to change script in other PR. Could you allow this?

Of Course. You can do it. But please make `impl-aggregation` as base branch of the new PR.

**Comment by dairyo** [comment]
@yongpingchen 

I adapted throwing error.

> >  would like to change script in other PR. Could you allow this?
> Of Course. You can do it. But please make impl-aggregation as base branch of the new PR.

I would like to make impl-trait-api as base branch of the new PR. I will make the new PR after this PR is merged.

What do you think?

**Comment by yongpingchen** [comment]
> I would like to make impl-trait-api as base branch of the new PR. I will make the new PR after this PR is merged.

By this way, I can't check the test result of this PR. I suggest to merge the new commits( which fix the CI script) to branch `impl-aggregation`, so that I can know the small tests are really passed for this PR.

**Comment by dairyo** [comment]
@yongpingchen 

@naoki-akimoto san will make the PR for this branch.

**Comment by dairyo** [comment]
> @naoki-akimoto san will make the PR for this branch.

@naoki-akimoto I have finished this. Please implement ActionResult.swift.

**Pull request merged**